src/sqlfluff/core/parser/lexer.py

```diff
@@ -310,200 +288,226 @@ def lex(
-                    # Find the next enriched marker.
-                    for elem in new_segment_buff[idx + 1 :]:
-                        if isinstance(elem.pos_marker, EnrichedFilePositionMarker):
-                            next_pos = (
-                                elem.pos_marker.source_slice.start,
-                                elem.pos_marker.templated_slice.start,
-                            )
-                            break
-                    else:
-                        # We're at the end of the file.
-                        next_pos = (
-                            len(templated_file.source_str),
-                            len(templated_file.templated_str),
-                        )
-                else:
-                    # It's a point and so we don't need to find the next one.
-                    next_pos = prev_pos
-                # Enrich by proxy
-                new_segment_buff[idx].pos_marker = EnrichedFilePositionMarker(
-                    statement_index=new_segment_buff[idx].pos_marker.statement_index,
-                    line_no=new_segment_buff[idx].pos_marker.line_no,
-                    line_pos=new_segment_buff[idx].pos_marker.line_pos,
-                    char_pos=new_segment_buff[idx].pos_marker.char_pos,
-                    templated_slice=slice(prev_pos[1], next_pos[1]),
-                    source_slice=slice(prev_pos[0], next_pos[0]),
-                    is_literal=False,
-                    source_pos_marker=FilePositionMarker(
-                        new_segment_buff[idx].pos_marker.statement_index,
-                        prev_line,
-                        prev_line_pos,
-                        prev_pos[0],
-                    ),
                 )
+        return violations
 
-        lexer_logger.debug("Enriched Segments:")
-        for seg in new_segment_buff:
-            lexer_logger.debug(
-                "\tTmp: %s\tSrc: %s\tSeg: %s",
-                getattr(seg.pos_marker, "templated_slice", None),
-                getattr(seg.pos_marker, "source_slice", None),
-                seg,
-            )
+    @staticmethod
+    def lex_match(forward_string: str, lexer_matchers: List[StringMatcher]) -> LexMatch:
+        """Iteratively match strings using the selection of submatchers."""
+        elem_buff: List[LexedElement] = []
+        while True:
+            if len(forward_string) == 0:
+                return LexMatch(forward_string, elem_buff)
+            for matcher in lexer_matchers:
+                res = matcher.match(forward_string)
+                if res.elements:
+                    # If we have new segments then whoop!
+                    elem_buff += res.elements
+                    forward_string = res.forward_string
+                    # Cycle back around again and start with the top
+                    # matcher again.
+                    break
+                else:
+                    continue
```

Comment:
Does this line do anything? It seems like execution would go to the top of the `for` loop next regardless.